### PR TITLE
replace subject id with digest

### PR DESCRIPTION
**Pull request by wy65701436**
Since it has to support push subject and accessories in either order, it has to replace digest with id

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: api/v2.0/swagger.yaml [hunk 1]
```diff
@@ -9349,10 +9349,9 @@ definitions:
         format: int64
         description: The artifact id of the accessory
         x-omitempty: false
-      subject_artifact_id:
```
Comment (Vad1mo) [inline api/v2.0/swagger.yaml:9352]: This is a breaking change, removing functionality. 
I would prefer to support both options.

**Review by Vad1mo — CHANGES_REQUESTED** [review]

**Review by chlins — APPROVED** [review]
lgtm

File: api/v2.0/swagger.yaml (hunk 1 above)
Comment (wy65701436) [reply api/v2.0/swagger.yaml:9352]: Okay, that's a good point. I will submit another pull request to reinstate it. However, we should consider keeping it in, as subsequent pull requests depend on it.

**Review by stonezdj — APPROVED** [review]

**Comment by stonezdj** [comment]
Is it possible to use artifact_digest instead of artifact_id in the artifact_accessory table? it could avoid joining the table when querying an artifact's accessory?

**Review by Vad1mo — APPROVED** [review]

**Pull request merged**